flink-autoscaler/src/main/java/org/apache/flink/autoscaler/metrics/ScalingMetrics.java

```diff
@@ -221,7 +221,7 @@ private static double getNumRecordsInPerSecond(
             numRecordsInPerSecond =
                     flinkMetrics.get(FlinkMetric.SOURCE_TASK_NUM_RECORDS_IN_PER_SEC);
         }
-        if (isSource && (numRecordsInPerSecond == null || numRecordsInPerSecond.getSum() == 0)) {
+        if (!isSource && (numRecordsInPerSecond == null || numRecordsInPerSecond.getSum() == 0)) {
             numRecordsInPerSecond =
                     flinkMetrics.get(FlinkMetric.SOURCE_TASK_NUM_RECORDS_OUT_PER_SEC);
         }
```

Comment:
Thanks @Yang-LI-CS for the PR!

These 2 conditions are indeed the same, it maybe unexpected intuitively. Maybe here logic is numRecordsInPerSecond from 3 metrics:

1. FlinkMetric.NUM_RECORDS_IN_PER_SEC
2. FlinkMetric.SOURCE_TASK_NUM_RECORDS_IN_PER_SEC
3. FlinkMetric.SOURCE_TASK_NUM_RECORDS_OUT_PER_SEC

When 1 is unavailable, get it from 2. When 2 is unavailable, get it from 3.

I'm not sure about it. Need to confirm with @mxm . 

- If my guess is right, it's not a bug. And I suggest this PR add a little comments to clarify it. 
- If my guess is wrong, it's a bug.  We can fix it.

WDYT?

Reply:
By the way, current change doesn't make sense.

When a task isn't source, it doesn't have `SOURCE_TASK_NUM_RECORDS_OUT_PER_SEC` metric.

Reply:
@1996fanrui thanks, I'll add some comments and do as your suggestion 👍

Reply:
@1996fanrui This time I have just removed the unused if condition and add 1 comment.

To be sure, let's wait feedback from @mxm

Reply:
I have just add some comments to clarify this at last 🙏